packages/grafana-runtime/src/services/SidecarService_EXPERIMENTAL.ts

```diff
@@ -19,18 +17,16 @@ interface Options {
  * @experimental
  */
 export class SidecarService_EXPERIMENTAL {
-  private _activePluginId: BehaviorSubject<string | undefined>;
   private _initialContext: BehaviorSubject<unknown | undefined>;
-  private localStorageKey: string | undefined;
+  private memoryLocationService: LocationService;
+  private history: LocationStorageHistory;
 
-  constructor(options: Options) {
-    this.localStorageKey = options.localStorageKey;
-    let initialId = undefined;
-    if (this.localStorageKey) {
-      initialId = localStorage.getItem(this.localStorageKey) || undefined;
-    }
-    this._activePluginId = new BehaviorSubject<string | undefined>(initialId);
+  constructor() {
     this._initialContext = new BehaviorSubject<unknown | undefined>(undefined);
+    // We need a local ref for this so we can tap into the location changes and drive rerendering of components based
+    // on it without having a parent Router.
+    this.history = createLocationStorageHistory({ storageKey: 'grafana.sidecar.history' });
```

Comment:
Isn't there a risk of the sidecar history getting out of sync with the Grafana's history due to a separate history used here?

Reply:
I mean the browser history inside Grafana, which is afaik controlled by the locationSrv.

Reply:
At this moment they are separate. Like the browser navigation right now controls only the main view. So if you go back the sidecar stays open, even if you didn't have it open on that URL.

I would consider this sort of MVP right now. In the future I think there will be 2 modes (depending on the app) which will be:
1. have you sidecar app state part of the URL so forward/backward and URL sharing will work as you would expect. This will probably be how explore apps will work because it makes sense to share a view from 2 separate apps.
2. the only in memory state like it is implemented here so far. This makes sense for something like the Investigations  app or something like the query library, where it does not make sense to share that state with other people (because that state is tied to the user) and it make sense for that app to follow you around and you continually interacting with it (like adding stuff to the Investigations app as you are moving around).